api/hypershift/v1alpha1/nodepool_types.go

```diff
@@ -869,12 +869,44 @@ type AgentNodePoolPlatform struct {
 	AgentLabelSelector *metav1.LabelSelector `json:"agentLabelSelector,omitempty"`
 }
 
+type AzureVMImageType string
+
+const (
+	ImageID          AzureVMImageType = "ImageID"
+	AzureMarketplace AzureVMImageType = "AzureMarketplace"
+)
+
+// AzureVMImage represents the different types of image data that can be provided for an Azure VM.
+// +union
+type AzureVMImage struct {
+	// Type is the type of image data that will be provided to the Azure VM. This can be either "ImageID" or
+	// "MarketplaceImageInfo".
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum:=ImageID;AzureMarketplace
+	// +unionDiscriminator
+	Type AzureVMImageType `json:"azureImageType"`
+
+	// ImageID is the Azure resource ID of a VHD image to use to boot the Azure VMs from.
+	//
+	// +optional
+	// +unionMember, optional
+	ImageID *AzureImageInfo `json:"imageID,omitempty"`
+
+	// MarketplaceImageInfo contains the Azure Marketplace image info to use to boot the Azure VMs from.
+	//
+	// +optional
+	// +unionMember, optional
+	MarketplaceImageInfo *MarketplaceImage `json:"marketplaceImage,omitempty"`
+}
+
 type AzureNodePoolPlatform struct {
 	VMSize string `json:"vmsize"`
-	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used:
-	// subscription/$subscriptionID/resourceGroups/$resourceGroupName/providers/Microsoft.Compute/images/rhcos.x86_64.vhd
-	// +optional
-	ImageID string `json:"imageID,omitempty"`
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Image is immutable"
+	// +kubebuilder:validation:Required
+	// +immutable
+	// +required
+	Image *AzureVMImage `json:"image,omitempty"`
```

Comment:
```suggestion
	Image AzureVMImage `json:"image"`
```
why is this a pointer with `omitempty` if its required?